pkg/infrastructure/aws/clusterapi/iam.go

```diff
@@ -0,0 +1,189 @@
+						"elasticloadbalancing:CreateListener",
+						"elasticloadbalancing:CreateLoadBalancer",
+						"elasticloadbalancing:CreateLoadBalancerPolicy",
+						"elasticloadbalancing:CreateLoadBalancerListeners",
+						"elasticloadbalancing:CreateTargetGroup",
+						"elasticloadbalancing:ConfigureHealthCheck",
+						"elasticloadbalancing:DeleteListener",
+						"elasticloadbalancing:DeleteLoadBalancer",
+						"elasticloadbalancing:DeleteLoadBalancerListeners",
+						"elasticloadbalancing:DeleteTargetGroup",
+						"elasticloadbalancing:DeregisterInstancesFromLoadBalancer",
+						"elasticloadbalancing:DeregisterTargets",
+						"elasticloadbalancing:Describe*",
+						"elasticloadbalancing:DetachLoadBalancerFromSubnets",
+						"elasticloadbalancing:ModifyListener",
+						"elasticloadbalancing:ModifyLoadBalancerAttributes",
+						"elasticloadbalancing:ModifyTargetGroup",
+						"elasticloadbalancing:ModifyTargetGroupAttributes",
+						"elasticloadbalancing:RegisterInstancesWithLoadBalancer",
+						"elasticloadbalancing:RegisterTargets",
+						"elasticloadbalancing:SetLoadBalancerPoliciesForBackendServer",
+						"elasticloadbalancing:SetLoadBalancerPoliciesOfListener",
+						"kms:DescribeKey",
+					},
+					Resource: iamv1.Resources{
+						"*",
+					},
+				},
+			},
+		},
+		"worker": {
+			Version: "2012-10-17",
+			Statement: []iamv1.StatementEntry{
+				{
+					Effect: "Allow",
+					Action: iamv1.Actions{
+						"ec2:DescribeInstances",
+						"ec2:DescribeRegions",
+					},
+					Resource: iamv1.Resources{"*"},
+				},
+			},
+		},
+	}
+)
+
+// putIAMRoles creates the roles used by control-plane and compute nodes.
+func putIAMRoles(clusterID string, ic *installconfig.InstallConfig) error {
+	logrus.Infoln("Creating IAM roles for control-plane and compute nodes")
+	// Create the IAM Role with the aws sdk.
+	// https://docs.aws.amazon.com/sdk-for-go/api/service/iam/#IAM.CreateRole
+	session, err := ic.AWS.Session(context.TODO())
+	if err != nil {
+		return fmt.Errorf("failed to load AWS session: %w", err)
+	}
+	svc := iam.New(session)
+
+	// Create the IAM Roles for master and workers.
+	clusterOwnedIAMTag := &iam.Tag{
+		Key:   aws.String(fmt.Sprintf("kubernetes.io/cluster/%s", clusterID)),
+		Value: aws.String("owned"),
+	}
+	assumePolicy := &iamv1.PolicyDocument{
+		Version: "2012-10-17",
+		Statement: iamv1.Statements{
+			{
+				Effect: "Allow",
+				Principal: iamv1.Principals{
+					iamv1.PrincipalService: []string{
+						"ec2.amazonaws.com",
+					},
+				},
+				Action: iamv1.Actions{
+					"sts:AssumeRole",
+				},
+			},
+		},
+	}
+	assumePolicyBytes, err := json.Marshal(assumePolicy)
+	if err != nil {
+		return fmt.Errorf("failed to marshal assume policy: %w", err)
+	}
+
+	for _, role := range []string{"master", "worker"} {
+		roleName := aws.String(fmt.Sprintf("%s-%s-role", clusterID, role))
+		if _, err := svc.GetRole(&iam.GetRoleInput{RoleName: roleName}); err != nil {
```

Comment:
Let's use the `WithContext` version of this function: 
```suggestion
		if _, err := svc.GetRoleWithContext(ctx, &iam.GetRoleInput{RoleName: roleName}); err != nil {
```